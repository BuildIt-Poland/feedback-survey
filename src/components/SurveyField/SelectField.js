import React from 'react';
import { FormField } from 'grommet';

import CustomSelect from '../CustomSelect';

const SelectField = function({ field, options }) {
  if (!field) {
    return null;
  }
  return (
    <FormField
      name={field.id.toString()}
      label={field.content}
      required={field.required}
      component={CustomSelect}
      options={options}
      placeholder="Select"
    />
  );
};

export default SelectField;
