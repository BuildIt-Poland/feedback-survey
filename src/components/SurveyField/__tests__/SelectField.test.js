import React from 'react';
import { create } from 'react-test-renderer';

import SelectField from '../SelectField';

describe('COMPONENT - SurveyField SelectField', () => {
  it('does not render without field', () => {
    const props = {
      field: {
        id: '123',
        type: 'select',
        content: 'Question content',
        required: false
      },
      options: ['One', 'Two']
    };
    const component = create(<SelectField {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
