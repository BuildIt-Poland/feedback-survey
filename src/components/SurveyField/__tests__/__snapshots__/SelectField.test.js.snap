// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField SelectField does not render without field 1`] = `
<div
  className="sc-bdVaJa fIyeZW FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
>
  <label
    className="StyledText-sc-1sadyjn-0 bMXxti"
  >
    Question content
  </label>
  <div
    className="StyledBox-sc-13pk1d4-0 gZSidW"
  />
</div>
`;
